clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -495,8 +497,22 @@ public ApiException build(String message) {
             public ApiException build(String message) {
                 return new OperationNotAttemptedException(message);
             }
-        });
-
+    }),
+    KAFKA_STORAGE_ERROR(56, "Disk error when trying to access log file on the disk.",
+        new ApiExceptionBuilder() {
+            @Override
+            public ApiException build(String message) {
+                return new KafkaStorageException(message);
+            }
+    }),
+    UNKNOWN_ERROR_CODE(57, "The client received an unknown error code when processing the response. This may happen if " +
+            "the client library version is lower than the server library version",
+        new ApiExceptionBuilder() {
```

Comment:
Thanks much for the quick review!

I think the current approach is probably simpler. The broker only needs to know 2 types of the clients instead of three, i.e. one that knows UnknownErrorCodeException and one that doesn't. If the request of the client suggests that the client knows UnknownErrorCodeException, then the broker will simply send the origin error code. The client library will convert the error to UnknownErrorCodeException is error code is not recognized. Otherwise, the broker should convert the new error code to an existing error code before sending the response. Note that we need at most one `if/else` in each response to check whether the client knows UnknownErrorCodeException given the request version.

On the other hand, the alternative requires Kafka to potentially have multiple `if/else` in each response to check whether the clients know each newly-added error code. The number of check will increase overtime as we add more and more new error code. Thus the current approach seems simpler. What do you think?